src/reactviews/pages/QueryResult/table/table.ts

```diff
@@ -210,18 +219,29 @@ export class Table<T extends Slick.SlickData> implements IThemable {
      * @returns true if filters were successfully loaded and applied, false if no filters were found
      */
     public async setupFilterState(): Promise<boolean> {
-        this.columns.forEach((column) => {
-            if (column.field) {
-                const filters =
-                    this.queryResultState.state.filterState[column.field];
-                if (filters) {
-                    (<FilterableColumn<T>>column).filterValues =
-                        filters.filterValues;
-                } else {
-                    return false;
+        const filterMapArray = store.get(
+            this.queryResultState.state.uri!,
+        ) as GridColumnMap[];
+        if (!filterMapArray) {
+            return false;
+        }
+        const filterMap = filterMapArray.find((filter) => filter[this.gridId]);
+        if (!filterMap || !filterMap[this.gridId]) {
+            console.log("No filters found in store");
+            return false;
+        }
+        for (const column of this.columns) {
+            for (const columnFilterMap of filterMap[this.gridId]) {
+                for (const columnId in columnFilterMap) {
```

Comment:
Thrice-nested for-loops are a bit sus; are they all necessary, or can this be optimized?  Are `this.columns` all in the same grid or spread across different grids?  Maybe a comment here explaining what's the loops are meant to do would help me understand better.

Reply:
Reworked to remove an unnecessary loop